abci/client/grpc_client.go

```diff
@@ -155,8 +155,8 @@ func (cli *grpcClient) Error() error {
 // NOTE: The callback may receive internally generated flush responses.
 func (cli *grpcClient) SetResponseCallback(resCb Callback) {
 	cli.mtx.Lock()
+	defer cli.mtx.Unlock()
```

Comment:
sorry, but this doesn't improve anything.

Reply:
Oh, i got it. I will change this parts

Reply:
@melekes Of course, in this situation, it’s fine not to use defer for unlocking the mutex. However, I believe it’s recommended to use defer whenever possible when unlocking a mutex.

Anyway, thanks for the review. I’ve removed the defer code from the PR.